### PR TITLE
blog: add DMP Week 4 Nirav Sharma

### PR DESCRIPTION
**Pull request by 021nirav-blip**
Week 4 report is up. This one was a big week – two PRs landed and the audio engine is finally solid.

PR 5.2 fixed the audio engine EDO issues: the A0 offset bug that turned pitch 0 into G9 in 5-EDO (instead of C4), got temperamentChanged doing proportional mapping for all 12 note names, and added _lastTemperamentBuilt tracking to fix stale cache issues.

started making widgets EDO-aware - Pitch Slider, Tuner, and Sampler are now working with any EDO. Three down, three to go.

The last two widgets (modewidget and musickeyboard) are deferred - they have 34+ and 14+ hardcoded 12 instances respectively and need a full redesign, not just quick swaps.

Tests: all 6799 passing across 192 suites. 7 files changed, +413 lines added, −65 removed.

A few things I learned along the way: proportional mapping with Math.round(step * 12 / currentEDO) is the right pattern for EDO : name conversion; Tone.js mocks in tests make frequencies NaN if you're not careful; and the 1200-cent constant is EDO-independent - don't touch it.

### CODE REVIEW

**Comment by 021nirav-blip** [comment]
@pikurasa

**Review by pikurasa — CHANGES_REQUESTED** [review]
You must get into the habit of reviewing your work.

Things like this make me concerned:
Image

Screenshot From 2026-07-22 10-18-05

**Comment by pikurasa** [comment]
...When I fetch your branches, I see "f51e91c..9371e40  nirav-blog-post-week5 -> nirav/nirav-blog-post-week5" but no corresponding PR. Please make a PR for that, since it is suppose to be prior to this one.

**Comment by 021nirav-blip** [comment]
can you Re: the branch fetch - for this PR

**Comment by pikurasa** [comment]
@021nirav-blip I'm sort of confused because this seems to be published already. See: https://www.sugarlabs.org/news/developer-news/2026-07-03-dmp-26-niravsharma-week04 (This is identical to what you have in this PR for http://localhost:5173/news/developer-news/2026-07-03-dmp-26-niravsharma-week04)

What I don't see PRs for yet is weeks 3 and 4...

**Comment by 021nirav-blip** [comment]
@pikurasa 
i updated the context of the file with corrent PR and info

**Comment by pikurasa** [comment]
Screenshot From 2026-08-21 21-30-39

@021nirav-blip Why are there two week fours?

File: src/constants/MarkdownFiles/posts/2026-07-18-dmp-26-niravsharma-week04.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+description: "Week 4: PR #7807 fixes the synth pipeline. Unicode accidental normalization, numberToPitch bounds, and temperament-aware pitch caching."
+tags: "dmp26,sugarlabs,week04,niravsharma,musicblocks,temperament"
+image: "assets/Images/c4gt_DMP.webp"
+---
+
+<!-- markdownlint-disable -->
+
+# Weekly Blog Post, 2026
+
+**Contributor:** Nirav Sharma
```
Comment (pikurasa) [inline src/constants/MarkdownFiles/posts/2026-07-18-dmp-26-niravsharma-week04.md]: This section is not consistent with your other blog posts. Please make them consistent.

Comment (pikurasa) [reply src/constants/MarkdownFiles/posts/2026-07-18-dmp-26-niravsharma-week04.md]: Image  from https://www.sugarlabs.org/news/developer-news/2026-07-09-dmp-26-niravsharma-week05 is a good example.

File: src/constants/MarkdownFiles/posts/2026-07-18-dmp-26-niravsharma-week04.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+- In synthutils.js, _getFrequency uses getOctaveRatio() instead of a hardcoded 2, and the temperament widget's pitch calculations use the active EDO's pitch count instead of a hardcoded 12.
+Small things that came along
+- Added 17-EDO (equal17) to the temperament dictionaries — it was just missing.
+- Extended EQUIVALENTNATURALS to cover double sharps (𝄪), so D𝄪 maps to E, E𝄪 to F♯, and so on. This also fixes the D♯ → E♭ rewrite edge case Devin had hit.
+- Tests: 8 new tests for the EQUIVALENTNATURALS mappings, the accidental normalization, and the pitch wrapping, plus updated mocks — a little over 130 lines added across musicutils.test.js, synthutils.test.js, and logo.test.js.
+What's next
+
+PR #7807 is merged. A few call sites are now threaded with the temperament but still need proper handling — calculateInvert, drumStyle, inPitchStaircase, inMusicKeyboard, and justMeasuring — so the next PR cleans those up and moves on to the EDO-native consumers and Lilypond export work.
+
+Links
```
Comment (pikurasa) [inline src/constants/MarkdownFiles/posts/2026-07-18-dmp-26-niravsharma-week04.md]: Your headers are not consistent with your other blogposts. Please make them consistent.

**Review by pikurasa — CHANGES_REQUESTED** [review]

**Comment by 021nirav-blip** [comment]
@pikurasa
